CodeGen/src/AssemblyBuilderA64.cpp

```diff
@@ -575,12 +584,18 @@ void AssemblyBuilderA64::fadd(RegisterA64 dst, RegisterA64 src1, RegisterA64 src
 
         placeR3("fadd", dst, src1, src2, 0b11110'01'1, 0b0010'10);
     }
-    else
+    else if (dst.kind == KindA64::s)
     {
         LUAU_ASSERT(dst.kind == KindA64::s && src1.kind == KindA64::s && src2.kind == KindA64::s);
 
         placeR3("fadd", dst, src1, src2, 0b11110'00'1, 0b0010'10);
     }
+    else
+    {
+        LUAU_ASSERT(dst.kind == KindA64::q && src1.kind == KindA64::q && src2.kind == KindA64::q);
+
+        placeVR("fadd", dst, src1, src2, 0b0'01110'0'0'1, 0b11010'1);
```

Comment:
Now I want to check if the arm docs really have different assignments for Q and S/D :)
With different bit group separation no less -_-

Reply:
<img width="1228" alt="Screenshot 2024-02-15 at 9 19 53 AM" src="https://github.com/luau-lang/luau/assets/1106629/2e7c6e61-23b6-47d1-8c81-6e1c5a273731">

Reply:
<img width="387" alt="image" src="https://github.com/luau-lang/luau/assets/75688451/ab6f9028-2a10-46f5-aa2b-26bf4647d356">

Amazing consistency :)